File: frontend/src/constants/atsStatus.js
/**
 * ATS pipeline status enum. Single source of truth for candidate stages.
 */
export const ATS_STATUS = {
  APPLIED: 'APPLIED',
  SCREENING: 'SCREENING',
  SHORTLISTED: 'SHORTLISTED',
  INTERVIEW: 'INTERVIEW',
  REJECTED: 'REJECTED',
  HIRED: 'HIRED',
}

/** Ordered list for pipeline columns (excludes terminal states for main board). */
export const ATS_PIPELINE_ORDER = [
  ATS_STATUS.APPLIED,
  ATS_STATUS.SCREENING,
  ATS_STATUS.SHORTLISTED,
  ATS_STATUS.INTERVIEW,
  ATS_STATUS.HIRED,
  ATS_STATUS.REJECTED,
]

/** Statuses that appear as Kanban columns (active pipeline). */
export const ATS_KANBAN_STATUSES = [
  ATS_STATUS.APPLIED,
  ATS_STATUS.SCREENING,
  ATS_STATUS.SHORTLISTED,
  ATS_STATUS.INTERVIEW,
  ATS_STATUS.HIRED,
  ATS_STATUS.REJECTED,
]

export const ATS_STATUS_LABEL = {
  [ATS_STATUS.APPLIED]: 'Applied',
  [ATS_STATUS.SCREENING]: 'Screening',
  [ATS_STATUS.SHORTLISTED]: 'Shortlisted',
  [ATS_STATUS.INTERVIEW]: 'Interview',
  [ATS_STATUS.REJECTED]: 'Rejected',
  [ATS_STATUS.HIRED]: 'Hired',
}

/** Badge variant per status for candidate/HR UI (default | secondary | outline | destructive) */
export const ATS_STATUS_BADGE_VARIANT = {
  [ATS_STATUS.APPLIED]: 'secondary',
  [ATS_STATUS.SCREENING]: 'default',
  [ATS_STATUS.SHORTLISTED]: 'default',
  [ATS_STATUS.INTERVIEW]: 'default',
  [ATS_STATUS.REJECTED]: 'destructive',
  [ATS_STATUS.HIRED]: 'default',
}

File: backend/go-backend/internal/domain/enums/ats_status.go
package enums

// ATSStatus is the single source of truth for applicant tracking state.
type ATSStatus string

const (
	ATSApplied    ATSStatus = "APPLIED"
	ATSScreening  ATSStatus = "SCREENING"
	ATSShortlisted ATSStatus = "SHORTLISTED"
	ATSInterview  ATSStatus = "INTERVIEW"
	ATSRejected   ATSStatus = "REJECTED"
	ATSHired      ATSStatus = "HIRED"
)

// Valid returns true if s is a known status.
func (s ATSStatus) Valid() bool {
	switch s {
	case ATSApplied, ATSScreening, ATSShortlisted, ATSInterview, ATSRejected, ATSHired:
		return true
	default:
		return false
	}
}

// All returns all valid statuses in display order.
func AllATSStatuses() []ATSStatus {
	return []ATSStatus{ATSApplied, ATSScreening, ATSShortlisted, ATSInterview, ATSRejected, ATSHired}
}
